src/bridge/transactions/take_2.rs

```diff
@@ -69,43 +72,60 @@ impl PreSignedMusig2Transaction for Take2Transaction {
 }
 
 impl Take2Transaction {
-    pub fn new(context: &OperatorContext, input0: Input, input1: Input, input2: Input) -> Self {
+    pub fn new(
+        context: &OperatorContext,
+        input_0: Input,
+        input_1: Input,
+        input_2: Input,
+        input_3: Input,
+    ) -> Self {
         let mut this = Self::new_for_validation(
             context.network,
             &context.operator_public_key,
+            &context.operator_taproot_public_key,
             &context.n_of_n_taproot_public_key,
-            input0,
-            input1,
-            input2,
+            input_0,
+            input_1,
+            input_2,
+            input_3,
         );
 
-        this.sign_input1(context);
+        this.sign_input_1(context);
+        this.sign_input_3(context);
 
         this
     }
 
     pub fn new_for_validation(
         network: Network,
         operator_public_key: &PublicKey,
+        operator_taproot_public_key: &XOnlyPublicKey,
         n_of_n_taproot_public_key: &XOnlyPublicKey,
-        input0: Input,
-        input1: Input,
-        input2: Input,
+        input_0: Input,
+        input_1: Input,
+        input_2: Input,
+        input_3: Input,
     ) -> Self {
         let connector_0 = Connector0::new(network, n_of_n_taproot_public_key);
-        let connector_2 = Connector2::new(network, operator_public_key);
-        let connector_3 = Connector3::new(network, n_of_n_taproot_public_key);
+        let connector_4 = Connector4::new(network, operator_public_key);
+        let connector_5 = Connector5::new(network, n_of_n_taproot_public_key);
+        let connector_c = ConnectorC::new(network, operator_taproot_public_key);
+
+        let input_0_leaf = 0;
```

Comment:
This is leaf 1 of connector 0.

Reply:
Ahh, good catch. This one is weird. We converted this from p2wsh to p2tr so there's only 1 leaf. I made it two leaves to be consistent with the rest of the codebase.